### DISCUSSION TITLE
How to use safemode option?

### DISCUSSION THREAD

**Discussion by Winson-Huang** [async-profiler/async-profiler#797] [Q&A] [answered]
In profiler.sh there is a --safe-mode option and in [`arguments.cpp`](https://github.com/async-profiler/async-profiler/blob/32601bccd9e49adda9510a2ed79d142ac6ef0ff9/src/arguments.cpp#L84C8-L84C8) there is a safemode argument. But what is it exactly? How to use it? The relevant note is "disable stack recovery techniques", I searched it but didn't find anything useful.

**Comment by apangin** [comment, marked as answer]
This option is only for troubleshooting async-profiler stack walking bugs. It should never be used normally, and therefore intentionally left undocumented. Anyway, the option will become obsolete after #795.
